src/main/java/com/avispl/symphony/dal/infrastructure/management/polycom/lens/PolyLensCommunicator.java

```diff
@@ -0,0 +1,1222 @@
+		synchronized (aggregatedDeviceList) {
+			for (AggregatedDevice aggregatedDevice : aggregatedDeviceList) {
+				AggregatedDevice newClonedAggregatedDevice = new AggregatedDevice();
+				newClonedAggregatedDevice.setDeviceId(aggregatedDevice.getDeviceId());
+				newClonedAggregatedDevice.setDeviceModel(aggregatedDevice.getDeviceModel());
+				newClonedAggregatedDevice.setDeviceName(aggregatedDevice.getDeviceName());
+				newClonedAggregatedDevice.setSerialNumber(aggregatedDevice.getSerialNumber());
+				newClonedAggregatedDevice.setDeviceOnline(aggregatedDevice.getDeviceOnline());
+				newClonedAggregatedDevice.setType(aggregatedDevice.getType());
+				newClonedAggregatedDevice.setCategory(aggregatedDevice.getCategory());
+				newClonedAggregatedDevice.setDeviceMake(aggregatedDevice.getDeviceMake());
+
+				Map<String, String> oldStats = aggregatedDevice.getProperties();
+				List<AdvancedControllableProperty> controllableProperties = aggregatedDevice.getControllableProperties();
+				Map<String, String> newProperties;
+				newProperties = mapMonitoringProperty(oldStats);
+				if (newClonedAggregatedDevice.getDeviceOnline()) {
+					createControl(controllableProperties, newProperties);
+				}
+
+				newClonedAggregatedDevice.setProperties(newProperties);
+				newClonedAggregatedDevice.setControllableProperties(controllableProperties);
+				resultAggregatedDeviceList.add(newClonedAggregatedDevice);
+			}
+		}
+		return resultAggregatedDeviceList;
+	}
+
+	/**
+	 * add control property into controllableProperties
+	 *
+	 * @param controllableProperties controllableProperty list of aggregated device
+	 * @param stats stats of aggregated device
+	 */
+	private void createControl(List<AdvancedControllableProperty> controllableProperties, Map<String, String> stats) {
+		stats.put(PolyLensConstant.REBOOT_DEVICE, PolyLensConstant.EMPTY);
+		AdvancedControllableProperty restartButton = createButton(PolyLensConstant.REBOOT_DEVICE, PolyLensConstant.REBOOT, PolyLensConstant.REBOOTING,
+				PolyLensConstant.GRACE_PERIOD);
+		controllableProperties.add(restartButton);
+	}
+
+	/**
+	 * map monitoring property into stats of aggregated device
+	 *
+	 * @param oldStats stats from model mapping
+	 * @return stats after modify data
+	 */
+	private Map<String, String> mapMonitoringProperty(Map<String, String> oldStats) {
+		Map<String, String> newProperties = new HashMap<>();
+		String connections = oldStats.get(PolyLensConstant.CONNECTIONS);
+		String entitlements = oldStats.get(PolyLensConstant.ENTITLEMENTS);
+		List<Connection> connectionList;
+		List<Entitlement> entitlementList;
+		String group;
+		for (PolyLensAggregatedMetric property : PolyLensAggregatedMetric.values()) {
+			String name = property.getName();
+			switch (property) {
+				case MODEL:
+					newProperties.put(PolyLensConstant.MODEL_GROUP + PolyLensConstant.NAME, getDefaultValueForNullData(oldStats.get(PolyLensConstant.MODEL_NAME)));
```

Comment:
it should be PolyLensConstant.MODEL_GROUP.concat(PolyLensConstant.NAME). The concat() performance is similar to string buffer when merging strings less than 3 instance, and much faster than + operation